receiver/statsdreceiver/protocol/statsd_parser_test.go

```diff
@@ -1001,3 +1005,222 @@ func TestTimeNowFunc(t *testing.T) {
 	timeNow := timeNowFunc()
 	assert.NotNil(t, timeNow)
 }
+
+func TestStatsDParser_AggregateTimerWithHistogram(t *testing.T) {
+	timeNowFunc = func() time.Time {
+		return time.Unix(711, 0)
+	}
+	// It is easiest to validate data in tests such as this by limiting the
+	// histogram size to a small number and then setting the maximum range
+	// to test at scale 0, which is easy to reason about.  The tests use
+	// max size 10, so tests w/ a range of 2**10 appear below.
+
+	// Note: The structure here uses the mapping functions from
+	// OTel-Go sdk/metric/aggregator/exponential/mapping, which
+	// are the state of the OTel specification before the
+	// inclusivity adjustment here:
+	// https://github.com/open-telemetry/opentelemetry-specification/pull/2633,
+	// Tests that use exact powers-of-two will shift by one after
+	// this change which is the reason we avoid using exact powers
+	// of two (or anything near an exact power of two).
+
+	normalMapping := []TimerHistogramMapping{
+		{
+			StatsdType:   "timer",
+			ObserverType: "histogram",
+			Histogram: HistogramConfig{
+				MaxSize: 10,
+			},
+		},
+		{
+			StatsdType:   "histogram",
+			ObserverType: "histogram",
+			Histogram: HistogramConfig{
+				MaxSize: 10,
+			},
+		},
+	}
+
+	newPoint := func() (pmetric.Metrics, pmetric.ExponentialHistogramDataPoint) {
+		data := pmetric.NewMetrics()
+		ilm := data.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty()
+		m := ilm.Metrics().AppendEmpty()
+		m.SetName("expohisto")
+		ep := m.SetEmptyExponentialHistogram()
+		ep.SetAggregationTemporality(pmetric.MetricAggregationTemporalityDelta)
+		dp := ep.DataPoints().AppendEmpty()
+
+		dp.Attributes().PutString("mykey", "myvalue")
```

Comment:
This also needs to be `PutStr` I think
```suggestion
		dp.Attributes().PutStr("mykey", "myvalue")
```